store-and-retrieve-data/use-storagehub-sdk/authenticate-with-siwe-and-jwt.md

```diff
@@ -11,102 +11,33 @@ This guide shows how to sign in to a StorageHub Main Storage Provider (MSP) usin
-
-First, you'll need to set up the necessary clients to connect to the DataHaven network, which runs on a dual-protocol architecture (Substrate for core logic and EVM for compatibility).
-
-If you've already followed the [Create a Bucket](/store-and-retrieve-data/use-storagehub-sdk/create-a-bucket/){target=\_blank} guide, your clients may already be initialized. Review the placeholders at the bottom of the following snippet to see where you'll add logic in this guide, then skip ahead to [Authenticate Your Address via MSP](#authenticate-your-address-via-msp).
-
-Create an `index.ts` and add the following code:
-
-!!! note
-    The code below uses **DataHaven Testnet** configuration values, which include the **Chain ID**, **RPC URL**, **WSS URL**, **MSP URL**, and **token metadata**. If you’re running a **local devnet**, make sure to replace these with your local configuration parameters. You can find all the relevant **local devnet values** in the [Starter Kit](/store-and-retrieve-data/starter-kit/).
+Add the following code to your `index.ts` file:
 
 ```ts title="index.ts"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/authenticate-with-siwe-and-jwt/authenticate.ts:imports'
-
-async function run() {
-  --8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/authenticate-with-siwe-and-jwt/authenticate.ts:initialize-clients'
-
-  // --- Issue storage request logic ---
-  // **PLACEHOLDER FOR STEP 1: AUTHENTICATE YOUR ADDRESS VIA MSP**
-  // **PLACEHOLDER FOR STEP 2: GET PROFILE INFO FROM MSP**
-
-  // Disconnect the Polkadot API at the very end
-  await polkadotApi.disconnect();
-}
-
-await run();
+--8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/authenticate-with-siwe-and-jwt/authenticate.ts'
 ```
 
-## Authenticate Your Address via MSP
+As you can see, the `authenticateUser` helper method is being called from your `mspService.ts` file. This is the method responsible for checking your address's authentication status and authenticating your address via the MSP Client. 
 
-To check your address's authentication status and to authenticate your address using the MSP Client, add the following code:
-
-```ts title="// **PLACEHOLDER FOR STEP 1: AUTHENTICATE YOUR ADDRESS VIA MSP**"
---8<-- 'code/store-and-retrieve-data/use-storagehub-sdk/authenticate-with-siwe-and-jwt/authenticate.ts:authenticate-via-msp'
-```
+Within the `authenticateUser` method, the SDK's `mspClient.auth.SIWE` method is triggered, which produces a JWT token that serves as proof of authentication. Within `mspService.ts`, this token is passed into the `sessionProvider` const that is one of the two required params for the `MspClient.connect` method. When you connect to an MSP while `sessionProvider` contains a valid JWT token, you are allowed to trigger certain MSP methods you otherwise wouldn't be able to (such as `MspClient.auth.getProfile`, `MspClient.files.uploadFile`, `MspClient.info.getPaymentStreams`).
 
-Run the script:
+## Run Auth Script
 
```

Comment:
There should be at least a sentence before the command saying what you're doing, I know it's in the title, but still applies